static/app/components/modals/projectCreationModal.tsx

```diff
@@ -138,7 +136,7 @@ export default function ProjectCreationModal({
         <Fragment>
           <Subtitle>{t('Choose a Platform')}</Subtitle>
           <PlatformPicker
-            defaultCategory={category}
+            defaultCategory={platform?.category ?? defaultCategory}
```

Comment:
letting the PlatformPicker manage the category state by removing the `useEffect` , we can clean up the code here